src/main/java/com/google/research/bleth/simulator/BoardStateFactory.java

```diff
@@ -57,7 +57,7 @@ public static BoardState create(Board board, String simulationId, int round) {
         int colNum = board.getColNum();
         ArrayTable<Integer, Integer, ArrayList<String>> boardStateTable = createEmptyTable(rowNum, colNum);
         Multimap<Location, IAgent> agentsOnBoard = board.agentsOnBoard();
-        for (Location location : agentsOnBoard.keys()) {
+        for (Location location : agentsOnBoard.keySet()) {
```

Comment:
Nit: this seems like an independent fix.
Prefer to fix such issues in a separate pull-request, to make review easier.